cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -1200,6 +1207,77 @@ func generateMountsModeRecover(mst *initramfsMountsState) error {
 	return nil
 }
 
+func generateMountsModeFactoryReset(mst *initramfsMountsState) error {
+	// steps 1 and 2 are shared with install mode
+	model, snaps, err := generateMountsCommonInstallRecover(mst)
+	if err != nil {
+		return err
+	}
+	// get the disk that we mounted the ubuntu-seed partition from as a
+	// reference point for future mounts
+	disk, err := disks.DiskFromMountPoint(boot.InitramfsUbuntuSeedDir, nil)
+	if err != nil {
+		return err
+	}
+	// step 3: find ubuntu-save, unlock and mount, note that factory-reset
+	// mode only cares about ubuntu-save, as ubuntu-data and ubuntu-boot
+	// will be wiped anyway so we do not even bother looking up those
+	// partitions (which may be corrupted too, hence factory-reset was
+	// invoked)
+	machine, err := func() (machine *recoverModeStateMachine, err error) {
+		allowFallback := true
+		machine = newRecoverModeStateMachine(model, disk, allowFallback)
+		// start from looking up encrypted ubuntu-save and unlocking with the fallback key
+		machine.current = machine.unlockMaybeEncryptedAloneSaveFallbackKey
+		for {
+			finished, err := machine.execute()
+			// TODO: consider whether certain errors are fatal or not
+			if err != nil {
+				return nil, err
+			}
+			if finished {
+				break
+			}
+		}
+		return machine, nil
+	}()
+
+	if err != nil {
+		return err
+	}
+
+	if machine.degraded() {
```

Comment:
does it actually make sense to use degraded here? it seems to look at UbuntuData stuff which we don't care about actually and also to look at run key which we don't plan to use anyway? Instead, is there some state of UbuntuSave partition that should stop us from continuing here?

Reply:
Per discussion with @pedronis, we'll keep it but use a different name for the file.

Reply:
The file name has been changed.